app/src/main/java/org/simple/clinic/summary/PatientSummaryEffectHandler.kt

```diff
@@ -102,63 +104,69 @@ class PatientSummaryEffectHandler @AssistedInject constructor(
         .build()
   }
 
   private fun loadStatinPrescriptionCheckInfo(): ObservableTransformer<LoadStatinPrescriptionCheckInfo, PatientSummaryEvent> {
     return ObservableTransformer { effects ->
       effects
           .observeOn(schedulersProvider.io())
-          .flatMap { effect ->
-            val patient = effect.patient
-            val today = LocalDate.now(userClock)
-                .atStartOfDay()
-                .atZone(userClock.zone)
-                .toInstant()
-
-            bloodPressureRepository.hasBPRecordedToday(
-                patientUuid = patient.uuid,
-                today = today,
-            ).map {
-              Pair(patient, it)
-            }
-          }
-          .flatMap { (patient, hasBPRecordedToday) ->
-            val prescriptionsObservable = prescriptionRepository.newestPrescriptionsForPatient(patient.uuid)
-            prescriptionsObservable.map { prescriptions ->
-              Triple(patient, prescriptions, hasBPRecordedToday)
-            }
-          }
-          .map { (patient, prescriptions, hasBPRecordedToday) ->
-            val assignedFacility = getAssignedFacility(patient.assignedFacilityId).toNullable()
-            val medicalHistory = medicalHistoryRepository.historyForPatientOrDefaultImmediate(
-                defaultHistoryUuid = uuidGenerator.v4(),
-                patientUuid = patient.uuid
-            )
-
-            StatinPrescriptionCheckInfoLoaded(
-                age = patient.ageDetails.estimateAge(userClock = userClock),
-                isPatientDead = patient.status == PatientStatus.Dead,
-                hasBPRecordedToday = hasBPRecordedToday,
-                assignedFacility = assignedFacility,
-                medicalHistory = medicalHistory,
-                prescriptions = prescriptions
-            )
-          }
-    }
-  }
-
-  private fun loadCVDRisk(): ObservableTransformer<LoadCVDRisk, PatientSummaryEvent> {
-    return ObservableTransformer { effects ->
-      effects
-          .observeOn(schedulersProvider.io())
-          .map {
-            val cvdRisk = cvdRiskRepository.getCVDRiskImmediate(it.patientUuid)
-            CVDRiskLoaded(cvdRisk?.riskScore)
+          .flatMap {
+            patientRepository.patient(it.patientUuid)
+                .extractIfPresent()
+                .flatMap { patient ->
+                  val cvdRisk = cvdRiskRepository.getCVDRiskImmediate(it.patientUuid)
+                  val today = LocalDate.now(userClock)
+                      .atStartOfDay(userClock.zone)
+                      .toInstant()
+
+                  Observable.combineLatest(
+                      medicalHistoryRepository.hasMedicalHistoryForPatientChangedSince(
+                          patientUuid = patient.uuid,
+                          instant = cvdRisk?.timestamps?.updatedAt ?: today,
+                      ),
```

Comment:
I am just thinking, can't we just place a observable to get medical history for patient and then compare the `updatedAt` time with the CVD risk created time if it's not null in the update function? 

Do we need another query to do this? 🤔

Reply:
Thanks for the feedback. Sounds really good and implemented that